app/src/main/java/de/danoeh/antennapod/fragment/swipeactions/SwipeActions.java

```diff
@@ -0,0 +1,169 @@
+package de.danoeh.antennapod.fragment.swipeactions;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.graphics.Canvas;
+
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.Arrays;
+import java.util.List;
+
+import de.danoeh.antennapod.dialog.SwipeActionsDialog;
+import de.danoeh.antennapod.fragment.EpisodesFragment;
+import de.danoeh.antennapod.model.feed.FeedItem;
+import de.danoeh.antennapod.model.feed.FeedItemFilter;
+import de.danoeh.antennapod.view.viewholder.EpisodeItemViewHolder;
+import it.xabaras.android.recyclerview.swipedecorator.RecyclerViewSwipeDecorator;
+
+public class SwipeActions {
+    public static final String PREF_NAME = "SwipeActionsPrefs";
+    public static final String PREF_SWIPEACTION_RIGHT = "PrefsSwipeActionRight";
+    public static final String PREF_SWIPEACTION_LEFT = "PrefsSwipeActionLeft";
+
+    public static final int ADD_TO_QUEUE = 0;
+    public static final int MARK_UNPLAYED = 1;
+    public static final int START_DOWNLOAD = 2;
+    public static final int MARK_FAV = 3;
+    public static final int MARK_PLAYED = 4;
```

Comment:
Looks like this should also be a property of the actions themselves

Reply:
I think that should stay where the array of all actions, so if the order is changes one does not have to go to each action class (which will be forgotten anyway)

Reply:
Oh, so the IDs here are related to the order of the items in the list below? To be honest, I did not realize this. Why do they need to be numbered? Eg, couldn't every action have a `TAG` field like the Fragments and then there could be a method that, given a tag, returns the action? Then we also don't need to instantiate all actions in a static variable but only those that are actually used.

Reply:
I changed that, due to your request to make the actions in separate classes, please have a look at the code, removing the need for tags and string arrays, it's all in that static array, which reduces complexity.

Reply:
I changed that while creating a class for each SwipeAction, i think the clear advantage of this is that theres only this one array, theres a comment stating `indexes of swipeActions`. I guess they could be removed as well, currently they are only used for the default suggestion, but it might be confusing to store the indexes somewhere else than the array, might lead unexpected bugs if the array is changed.